### PR TITLE
Fix release.yaml and dockerfiles, that cause Trivy failures.

### PR DESCRIPTION
**Pull request by maneeshaxyz**
## Summary

Provide a brief description of what this PR accomplishes and why these changes are needed.

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] Performance improvement
- [ ] Other (please describe):

## Changes Made

- List the main changes made in this PR
- Be specific about what was modified, added, or removed
- Include any architectural changes

## Testing

- [x] I have tested this change locally
- [ ] I have added unit tests for new functionality
- [ ] I have tested edge cases
- [x] All existing tests pass

## Checklist

- [ ] My code follows the project's style guidelines
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have checked that there are no merge conflicts

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -213,14 +213,24 @@ jobs:
           output: trivy-${{ matrix.image_name }}.sarif
           severity: CRITICAL,HIGH
           exit-code: '1'
```
Comment (ginaxu1) [inline .github/workflows/release.yml]: you can also explicitly ignore the known upstream dependencies using a .trivyignore file if you're trying to prevent a specific third-party image from failing the build
```
    trivyignores: '.github/.trivyignore-migrate'
```

**Review by ginaxu1 — CHANGES_REQUESTED** [review]

**Comment by maneeshaxyz** [comment]
Trivy scans in the release pipeline were failing due to HIGH/CRITICAL CVEs. This PR addresses those failures so the container images pass security scanning.

The changes are being merged into the staging pipeline first to validate them end-to-end. Once verified, the same changes can be applied to the dev pipeline. The staging target is only for easier testing; this is fundamentally a dev pipeline fix.

Most vulnerabilities were resolved by updating base images and dependencies we control. However, the migrations image still reports multiple Go binary CVEs because it uses the external `migrate/migrate` image, which bundles an outdated `migrate` binary that we do not maintain.

For the frontend/nginx image, findings were reduced to a single remaining HIGH vulnerability:

* `CVE-2026-27135` in Alpine `nghttp2-libs`

The following were attempted:

* updating Alpine/nginx base images
* switching to `stable-alpine`
* running `apk upgrade`

The vulnerability persists because Alpine has not yet published the patched package (`1.68.1`) in the repository. Current recommendation is:

* remove `apk upgrade` to preserve reproducible builds
* temporarily ignore this specific CVE in Trivy until Alpine releases the fix

Testing workflow:

```bash
git tag v0.0.0
git push origin v0.0.0
```

Additional notes:

* If pushing from this branch, squash/fixup the last 3 commits
* Cancel the workflow in `nsw` if duplicate runs are triggered

**Review by sthanikan2000 — APPROVED** [review]
lgtm

**Review by ginaxu1 — APPROVED** [review]
